crates/formality-prove/src/decls.rs

```diff
@@ -125,12 +125,23 @@ pub struct NegImplDeclBoundData {
 }
 
 /// Mark a trait or trait impl as `unsafe`.
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub enum Safety {
     Safe,
     Unsafe,
 }
```

Comment:
I think you could get this same effect with 

```rust
#[term]
pub enum Safety {
    #[grammar()]
    Safe,
    #[grammar(unsafe)]
    Unsafe,
}
```

That said, I didn't recommend it earlier because I thought we probably wanted to keep `safe` in the debug output. I'm torn between:

* Accept (and print) both `safe` and `unsafe` but have a default when parsing -- what I was originally thinking
* Never print safe -- wht you are doing

The former seems more explicit and clear, but it departs a bit from Rust surface syntax. I still think that's the way to go though, we do similar things in a few other places. I do think it's important to not require users to write `safe trait` in tests though.